include/envoy/router/filter_config_provider.h

```diff
@@ -0,0 +1,86 @@
+#pragma once
+
+#include "envoy/config/core/v3/config_source.pb.h"
+#include "envoy/config/core/v3/extension.pb.h"
+#include "envoy/http/filter.h"
+#include "envoy/init/manager.h"
+#include "envoy/server/filter_config.h"
+
+namespace Envoy {
+namespace Router {
+
+/**
+ * A provider for constant filter configurations.
+ */
+class FilterConfigProvider {
+public:
+  virtual ~FilterConfigProvider() = default;
+
+  /**
+   * Get the filter config resource name.
+   **/
+  virtual const std::string& name() PURE;
+
+  /**
+   * @return Http::FilterFactoryCb a filter config to be instantiated on the subsequent streams.
+   * Note that if the provider has not yet performed an initial configuration load and no default is
+   * provided, no information will be returned.
+   */
+  virtual absl::optional<Http::FilterFactoryCb> config() PURE;
+
+  /**
+   * Validate if the route configuration can be applied in the context of the filter manager.
+   * @param Server::Configuration::NamedHttpFilterConfigFactory a filter factory to validate in the
+   * context of the filter manager filter chains.
+   */
```

Comment:
It's a bit unclear to me what this does. Are you talking about the per-route filter config?

Reply:
Sorry, this is about filter configuration checked whether it's terminal.